MenuFramework/ConsoleMenu.cs

```diff
@@ -291,6 +205,188 @@ public MenuOptionResult Show()
             }
         }
 
+        private MenuOption DrawMenuAndGetSelection_KeyStringMode(MenuOption selectedOption)
+        {
+            // Draw all the menu items
+            Console.BackgroundColor = config.ItemBackgroundColor;
+            Console.ForegroundColor = config.ItemForegroundColor;
+            Console.Clear();
+            OnBeforeShow();
+
+            // Print the options
+            int maxKeyStringLength = 0;
+            int nextSerialNumber = 1;
+            
+            // Loop once to build a dictionary and do some calculations
+            Dictionary<string, MenuOption> optionDictionary = new Dictionary<string, MenuOption>(StringComparer.OrdinalIgnoreCase);
+            foreach (MenuOption option in menuOptions)
+            {
+                // Figure out the KeyString for the item
+                string keyString = option.KeyString ?? nextSerialNumber++.ToString();
+
+                // Calculate the current Max length of all key strings
+                maxKeyStringLength = Math.Max(maxKeyStringLength, keyString.Length);
+
+                // Add the item to the dictionary (throw if duplicate)
+                if (optionDictionary.ContainsKey(keyString))
+                {
+                    throw new Exception($"A duplicate menu key-string value was detected: {keyString} ");
+                }
+                optionDictionary.Add(keyString, option);
+            }
+
+            // Now loop the dictionary to print the menu 
+            foreach(var kvp in optionDictionary)
+            {
+                string menuLine = $"{kvp.Key.PadLeft(maxKeyStringLength)}{config.KeyStringTextSeparator}{kvp.Value.Text}";
+                if (selectedOption == kvp.Value)
+                {
+                    // Print the selected item in the configured color
+                    Console.BackgroundColor = config.SelectedItemBackgroundColor;
+                    Console.ForegroundColor = config.SelectedItemForegroundColor;
+                    Console.WriteLine(menuLine);
+                    Console.BackgroundColor = config.ItemBackgroundColor;
+                    Console.ForegroundColor = config.ItemForegroundColor;
+
+                }
```

Comment:
How would this be used? To show the previously selected menu item? I've noticed in the Joke menu it works when returning from a Joke back to the joke menu but if I go to the forecast/weather menu it doesn't do anything since they're auto numbered. 

Should we get rid of it?

Reply:
It does work. It's just that the config of TodayMenu does not set the SelectedItemForegroundColor, so it does not appear different.  I don't care if we keep it.  I was just trying to overload the meaning if "selectedItem...Color" to the context of the keystroke menu, showing the *previously* selected item in that highlighted color.  The other thing it does is displays the "Select an Option:" prompt in that same color.  We can remove it if you like, but it's easy enough to "turn off" by NOT changing the SelectedItem...Color configuration.

BUT... It DOES NOT work if the option uses the RebuildMenuOptions override (because those are new instances of MenuOption).